### DISCUSSION TITLE
Custom values for Prusa for TSMC

### DISCUSSION THREAD

**Discussion by MidniteM** [sn4k3/UVtools#271] [Q&A] [answered]
Hi,

The release notes for 2.19.0 says:
(Add) PrusaSlicer printer notes variables:
BottomLiftHeight2
BottomLiftSpeed2
LiftHeight2
LiftSpeed2
BottomRetractSpeed
BottomRetractSpeed2
BottomRetractHeight2
BottomRetractSpeed2
RetractHeight2
RetractSpeed2

I've tried re-adding my printer profile to prusa slicer but it doesnt have the new TSMC variables, I've also tried to manually add them but none of the values seem to copy into the ctb v4 file. Do the custom values have to be in a specific order?

**Comment by sn4k3** [comment, marked as answer]
Yes, they wont work. Because at convertion time, CTB is always v3. It turn into v4 only after convertion (Encoding time). That will ignore those because v3 is not compatible with them. But i will patch it to allow that use and set v4 before convertion.

Also i dont recommend to anyone to use TSMC, is a waste of time. All you need is 1 lift sequence and tune accelaration to right value and use the minimum lift height as you can

**Reply by sn4k3** [reply]
Fixed on new version

**Reply by MidniteM** [reply]
Before the file is converted though - in Prusa under 'Printer Settings', 'Notes'
I have the following custom values:
START_CUSTOM_VALUES
LightOffDelay_0
BottomLightOffDelay_0
BottomLiftHeight_5
LiftHeight_5
BottomLiftSpeed_65
LiftSpeed_65
RetractSpeed_150
BottomLightPWM_255
LightPWM_255
END_CUSTOM_VALUES

Is there a way to add the 2nd speed/height settings in Prusa? The 'Install Profiles into PrusaSlicer' tool in UVtools isn't including the new values and I wasn't sure if it was supposed to by the release notes in 2.19.0?

TSMC is becoming more trouble than it's worth for sure - is there an easy way to edit the acceleration value?

**Reply by sn4k3** [reply]
Maybe you are skiping the FILEFORMAT? you need to set it to V4.CTB or else it will be v3...
Serious TSMC is useless BS! acceleration is set via Gcode to set EEPROM, you need to find the command for your printer.

**Comment by MidniteM** [comment]
yeah I think I'll just set the TSMC back to 0.

For what its worth, UVtools seems to be calculating the lift height incorrectly still.
With my below test it was supposed to lift 2mm at 40mm/min then faster for 5mm at 150mm/min. With the file sliced in Pruse/UVtools it seemed to lift about 3mm slowly then go straight to the retract motion.

Layers file from a chitubox slice:
[10]
NonZeroPixelCount: 2002126
BoundingRectangle: {X=1205,Y=365,Width=1430,Height=1430}
IsBottomLayer: False
LayerHeight: 0.03
PositionZ: 0.33
LightOffDelay: 0
WaitTimeBeforeCure: 0
ExposureTime: 2.75
WaitTimeAfterCure: 0
**LiftHeight: 7**
LiftSpeed: 40
**LiftHeight2: 5**
LiftSpeed2: 150
WaitTimeAfterLift: 0
RetractHeight: 11.5
RetractSpeed: 150
RetractHeight2: 0.5
RetractSpeed2: 30
LightPWM: 255
MaterialMilliliters: 0.0736ml (4.45%)

Same file/same settings from a Prusa slice converted with UVtools:
[10]
NonZeroPixelCount: 2002038
BoundingRectangle: {X=1205,Y=365,Width=1430,Height=1430}
IsBottomLayer: False
LayerHeight: 0.03
PositionZ: 0.33
LightOffDelay: 0
WaitTimeBeforeCure: 0
ExposureTime: 2.75
WaitTimeAfterCure: 0
**LiftHeight: 5**
LiftSpeed: 40
**LiftHeight2: 2**
LiftSpeed2: 150
WaitTimeAfterLift: 0
RetractHeight: 7
RetractSpeed: 150
RetractHeight2: 0
RetractSpeed2: 80
LightPWM: 255
MaterialMilliliters: 0.0736ml (5.15%)

I hope this helps.

**Reply by sn4k3** [reply]
The properties are correct:

```
LiftHeight: 7
LiftSpeed: 40
LiftHeight2: 5
LiftSpeed2: 150
```

7-5 = 2mm for first lift and 5mm for the second lift at the noted speeds.

Can you send me the final file (ctb) you used to do this test? And also a full screenshot of UVtools main window with the file open?

**Reply by MidniteM** [reply]
I agree, the properties appear correct but do not behave correctly.

files attached :)

[testfiles.zip](https://github.com/sn4k3/UVtools/files/7035596/testfiles.zip)

**Reply by sn4k3** [reply]
The two files, one works other dont right?
I guess original chitubox file works while prusa dont?

**Reply by MidniteM** [reply]
correct, the chitubox slice works as expected.

**Reply by sn4k3** [reply]
OK lets go by steps, try this:

[TestPrusa.v4.zip](https://github.com/sn4k3/UVtools/files/7035755/TestPrusa.v4.zip)

**Reply by MidniteM** [reply]
The normal layers still dont work - lifts at 40mm/min for approx 4 seconds, then retracts for maybe 0.5 seconds at 150mm/min.

**Reply by sn4k3** [reply]
Ok test these by order:

[3-TestPrusa.v4.zip](https://github.com/sn4k3/UVtools/files/7039962/3-TestPrusa.v4.zip)

**Reply by sn4k3** [reply]
To discard TSMC and use accelaration the gcode you need is: `M8008 I<value>;`
By default i think is set to 800, you can try 400:  `M8008 I400;` and then you can use high speed values everywhere eg 200mm/min

in end of .gcode file you must have `M8500;` to save the values permanently

**Reply by sn4k3** [reply]
This should be fixed on v2.20.2. Please try

**Reply by MidniteM** [reply]
Sorry I havent had much time for testing lately - I'm using v2.20.2 now and everything seems to be working perfectly.

And thanks for the gcode values, next time I can be bothered playing with it I'll give the acceleration a try for sure :)

cheers!
